src/greenpak/i2c.py

```diff
@@ -185,3 +266,24 @@ def read(
                 )
                 print(str)
         return None
+
+    @override
+    def write_read(self, addr: int, wreq: bytearray, rbytes: int,
```

Comment:
Please add to this SMB implementation also gp_read/write/scan methods.

Reply:
It is in my pipeline to address the changes needed per your wishes.
It requires to change / re-work reads, writes, scan of SMBus adapter.